### ISSUE TITLE
ON_KERNEL_ANTI_ACTION reads empty list

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#1337]
`FuncON_KERNEL_ANTI_ACTION`, at trans.c:1497, starts by doing `INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0`. However, if `ker` is an empty list this then tries to read the 0th element of an empty list.

I think this just needs changing to something like `if (LEN_LIST(ker) == 0 || INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {`, but I'm not sure exactly what the semantics of this function should be, could you take a look @james-d-mitchell ?

**Comment by james-d-mitchell** [comment]
I see your point, but the suggested fix isn't quite right. If the argument is an empty list, then it should just return that empty list. However, I'm not sure there is any point in making this change, since this is not a user function, and the case of the empty list is not allowed in the GAP level version `OnKernelAntiAction`. On the other hand, it probably won't do any harm either.

**Comment by ChrisJefferson** [comment]
This function does get called with an empty list by some of the trans tests (that's how I came across it). I can track down which one if you like.

**Comment by james-d-mitchell** [comment]
Then it better be changed by adding:

    if (LEN_LIST(ker) == 0) {
      return ker;
    } else if (LEN_LIST(ker) == 1 && INT_INTOBJ(ELM_LIST(ker, LEN_LIST(ker))) == 0) {
    ...

**Comment by james-d-mitchell** [comment]
I was about to make a pull request to address this, but when I run `tst/testinstall/trans.tst` the line `return ker;` in the above is not actually executed. Was is a test in a different file @ChrisJefferson?

**Comment by james-d-mitchell** [comment]
The code is in [this branch](https://github.com/james-d-mitchell/gap/tree/on-kernel-anti-action) of my fork.

**Comment by ChrisJefferson** [comment]
It's done in semigroups 2.8.0, attributes.tst. You can hit it with this (the `Size(MinimalIdeal(s))` causes it:

```
gap> t := Transformation( [ 1 ] );;

# Trivial full transformation monoid T_1
# Previously this crashed: see issue #121 on Bitbucket
gap> s := Semigroup(t); # with displaying the semigroup
<trivial transformation group of degree 0 with 1 generator>
gap> MultiplicativeZero(s) = t;
true
gap> Size(MinimalIdeal(s)) = 1;
true
```

**Comment by james-d-mitchell** [comment]
Ah, ok, I think I made the wrong fix in my branch, I'll updated it and make a PR later today.

**Comment by ChrisJefferson** [comment]
Did you ever produce a fix for this?

**Comment by james-d-mitchell** [comment]
I thought so, will check later today

**Comment by fingolfin** [comment]
What's the status of this?

**Comment by fingolfin** [comment]
I suspect @james-d-mitchell fixed this in PR #1424 resp. 9326bbad3d5e6c230377c0784961446d74c5d4cd ... ?

**Comment by james-d-mitchell** [comment]
I'll check tomorrow morning

**Comment by james-d-mitchell** [comment]
This is resolved by PR #1424 as @fingolfin says.
